// @mintlify/validation/dist/mint-config/schemas/v2/properties/navigation/anchors.js
import { z } from 'zod';
import { colorSchemaWithOptionalLightAndDark } from '../reusable/color.js';
import { hiddenSchema } from '../reusable/hidden.js';
import { hrefSchema } from '../reusable/href.js';
import { iconSchema } from '../reusable/icon.js';
import { openApiSchema } from '../reusable/openapi.js';
import { decoratedDropdownsSchema, dropdownsSchema } from './dropdown.js';
import { decoratedGroupsSchema, groupsSchema } from './groups.js';
import { decoratedLanguagesSchema, languagesSchema } from './languages.js';
import { decoratedPagesSchema, pagesSchema } from './pages.js';
import { decoratedTabsSchema, tabsSchema } from './tabs.js';
import { decoratedVersionsSchema, versionsSchema } from './version.js';
export const baseAnchorSchema = z.object({
    anchor: z.string().nonempty().describe('The name of the anchor'),
    icon: iconSchema.optional(),
    color: colorSchemaWithOptionalLightAndDark.optional(),
    hidden: hiddenSchema.optional(),
});
export const anchorSchema = z.union([
    baseAnchorSchema.extend({ href: hrefSchema, openapi: openApiSchema.optional() }),
    baseAnchorSchema.extend({
        languages: z.lazy(() => languagesSchema),
        openapi: openApiSchema.optional(),
    }),
    baseAnchorSchema.extend({
        versions: z.lazy(() => versionsSchema),
        openapi: openApiSchema.optional(),
    }),
    baseAnchorSchema.extend({
        dropdowns: z.lazy(() => dropdownsSchema),
        openapi: openApiSchema.optional(),
    }),
    baseAnchorSchema.extend({
        tabs: z.lazy(() => tabsSchema),
        openapi: openApiSchema.optional(),
    }),
    baseAnchorSchema.extend({
        groups: z.lazy(() => groupsSchema),
        openapi: openApiSchema.optional(),
    }),
    baseAnchorSchema.extend({
        pages: z.lazy(() => pagesSchema),
        openapi: openApiSchema.optional(),
    }),
    baseAnchorSchema.extend({ openapi: openApiSchema }),
]);
export const decoratedAnchorSchema = baseAnchorSchema.and(z.union([
    z.object({ href: hrefSchema }),
    z.lazy(() => z.object({ languages: decoratedLanguagesSchema })),
    z.lazy(() => z.object({ versions: decoratedVersionsSchema })),
    z.lazy(() => z.object({ dropdowns: decoratedDropdownsSchema })),
    z.lazy(() => z.object({ tabs: decoratedTabsSchema })),
    z.lazy(() => z.object({ groups: decoratedGroupsSchema })),
    z.lazy(() => z.object({ pages: decoratedPagesSchema })),
]));
export const nonRecursiveAnchorSchema = baseAnchorSchema.extend({ href: hrefSchema });
export const anchorsSchema = z
    .array(anchorSchema)
    .min(1, 'At least one anchor must be specified')
    .describe('Organizing by anchors');
export const decoratedAnchorsSchema = z
    .array(decoratedAnchorSchema)
    .describe('Organizing by anchors');
