// @mintlify/validation/dist/mint-config/schemas/v2/properties/navigation/dropdown.js
import { z } from 'zod';
import { colorSchemaWithOptionalLightAndDark } from '../reusable/color.js';
import { hiddenSchema } from '../reusable/hidden.js';
import { hrefSchema } from '../reusable/href.js';
import { iconSchema } from '../reusable/icon.js';
import { openApiSchema } from '../reusable/openapi.js';
import { anchorsSchema, decoratedAnchorsSchema } from './anchors.js';
import { decoratedGroupsSchema, groupsSchema } from './groups.js';
import { decoratedLanguagesSchema, languagesSchema } from './languages.js';
import { decoratedPagesSchema, pagesSchema } from './pages.js';
import { decoratedTabsSchema, tabsSchema } from './tabs.js';
import { decoratedVersionsSchema, versionsSchema } from './version.js';
export const baseDropdownSchema = z.object({
    dropdown: z.string().nonempty().describe('The name of the dropdown'),
    icon: iconSchema.optional(),
    color: colorSchemaWithOptionalLightAndDark.optional(),
    description: z.string().optional().describe('The description of the dropdown'),
    hidden: hiddenSchema.optional(),
});
export const dropdownSchema = z.union([
    baseDropdownSchema.extend({ href: hrefSchema, openapi: openApiSchema.optional() }),
    baseDropdownSchema.extend({
        languages: z.lazy(() => languagesSchema),
        openapi: openApiSchema.optional(),
    }),
    baseDropdownSchema.extend({
        versions: z.lazy(() => versionsSchema),
        openapi: openApiSchema.optional(),
    }),
    baseDropdownSchema.extend({
        tabs: z.lazy(() => tabsSchema),
        openapi: openApiSchema.optional(),
    }),
    baseDropdownSchema.extend({
        anchors: z.lazy(() => anchorsSchema),
        openapi: openApiSchema.optional(),
    }),
    baseDropdownSchema.extend({
        groups: z.lazy(() => groupsSchema),
        openapi: openApiSchema.optional(),
    }),
    baseDropdownSchema.extend({
        pages: z.lazy(() => pagesSchema),
        openapi: openApiSchema.optional(),
    }),
    baseDropdownSchema.extend({ openapi: openApiSchema }),
]);
export const decoratedDropdownSchema = baseDropdownSchema.and(z.union([
    z.object({ href: hrefSchema }),
    z.lazy(() => z.object({ languages: decoratedLanguagesSchema })),
    z.lazy(() => z.object({ versions: decoratedVersionsSchema })),
    z.lazy(() => z.object({ tabs: decoratedTabsSchema })),
    z.lazy(() => z.object({ anchors: decoratedAnchorsSchema })),
    z.lazy(() => z.object({ groups: decoratedGroupsSchema })),
    z.lazy(() => z.object({ pages: decoratedPagesSchema })),
]));
export const nonRecursiveDropdownSchema = baseDropdownSchema.extend({ href: hrefSchema });
export const dropdownsSchema = z.array(dropdownSchema).describe('Organizing by dropdowns');
export const decoratedDropdownsSchema = z
    .array(decoratedDropdownSchema)
    .min(1, 'At least one dropdown must be specified')
    .describe('Organizing by dropdowns');
