// @mintlify/validation/dist/mint-config/schemas/v2/properties/navigation/languages.js
import { z } from 'zod';
import { hiddenSchema } from '../reusable/hidden.js';
import { hrefSchema } from '../reusable/href.js';
import { openApiSchema } from '../reusable/openapi.js';
import { anchorsSchema, decoratedAnchorsSchema } from './anchors.js';
import { decoratedDropdownsSchema, dropdownsSchema } from './dropdown.js';
import { decoratedGroupsSchema, groupsSchema } from './groups.js';
import { decoratedPagesSchema, pagesSchema } from './pages.js';
import { decoratedTabsSchema, tabsSchema } from './tabs.js';
import { decoratedVersionsSchema, versionsSchema } from './version.js';
export const baseLanguageSchema = z.object({
    language: z
        .enum(['en', 'cn', 'es', 'fr', 'jp', 'pt', 'pt-BR', 'de'])
        .describe('The name of the language in the ISO 639-1 format'),
    default: z.boolean().optional().describe('Whether this language is the default language'),
    hidden: hiddenSchema.optional(),
});
export const languageSchema = z
    .union([
    baseLanguageSchema.extend({ href: hrefSchema, openapi: openApiSchema.optional() }),
    baseLanguageSchema.extend({
        versions: z.lazy(() => versionsSchema),
        openapi: openApiSchema.optional(),
    }),
    baseLanguageSchema.extend({
        tabs: z.lazy(() => tabsSchema),
        openapi: openApiSchema.optional(),
    }),
    baseLanguageSchema.extend({
        dropdowns: z.lazy(() => dropdownsSchema),
        openapi: openApiSchema.optional(),
    }),
    baseLanguageSchema.extend({
        anchors: z.lazy(() => anchorsSchema),
        openapi: openApiSchema.optional(),
    }),
    baseLanguageSchema.extend({
        groups: z.lazy(() => groupsSchema),
        openapi: openApiSchema.optional(),
    }),
    baseLanguageSchema.extend({
        pages: z.lazy(() => pagesSchema),
        openapi: openApiSchema.optional(),
    }),
    baseLanguageSchema.extend({ openapi: openApiSchema }),
])
    .describe('Organizing by languages');
export const decoratedLanguageSchema = baseLanguageSchema
    .and(z.union([
    z.object({ href: hrefSchema }),
    z.lazy(() => z.object({ versions: decoratedVersionsSchema })),
    z.lazy(() => z.object({ tabs: decoratedTabsSchema })),
    z.lazy(() => z.object({ dropdowns: decoratedDropdownsSchema })),
    z.lazy(() => z.object({ anchors: decoratedAnchorsSchema })),
    z.lazy(() => z.object({ groups: decoratedGroupsSchema })),
    z.lazy(() => z.object({ pages: decoratedPagesSchema })),
]))
    .describe('Organizing by languages');
export const nonRecursiveLanguageSchema = baseLanguageSchema.extend({ href: hrefSchema });
export const languagesSchema = z
    .array(languageSchema)
    .min(1, 'At least one language must be specified')
    .describe('Organizing by languages');
export const decoratedLanguagesSchema = z
    .array(decoratedLanguageSchema)
    .min(1, 'At least one language must be specified')
    .describe('Organizing by languages');
