// @mintlify/validation/dist/mint-config/schemas/v2/properties/navigation/version.js
import { z } from 'zod';
import { hiddenSchema } from '../reusable/hidden.js';
import { hrefSchema } from '../reusable/href.js';
import { openApiSchema } from '../reusable/openapi.js';
import { anchorsSchema, decoratedAnchorsSchema } from './anchors.js';
import { decoratedDropdownsSchema, dropdownsSchema } from './dropdown.js';
import { decoratedGroupsSchema, groupsSchema } from './groups.js';
import { decoratedLanguagesSchema, languagesSchema } from './languages.js';
import { decoratedPagesSchema, pagesSchema } from './pages.js';
import { decoratedTabsSchema, tabsSchema } from './tabs.js';
export const baseVersionSchema = z.object({
    version: z.string().nonempty().describe('The name of the version'),
    default: z.boolean().optional().describe('Whether this version is the default version'),
    hidden: hiddenSchema.optional(),
});
export const versionSchema = z.union([
    baseVersionSchema.extend({ href: hrefSchema, openapi: openApiSchema.optional() }),
    baseVersionSchema.extend({
        languages: z.lazy(() => languagesSchema),
        openapi: openApiSchema.optional(),
    }),
    baseVersionSchema.extend({
        tabs: z.lazy(() => tabsSchema),
        openapi: openApiSchema.optional(),
    }),
    baseVersionSchema.extend({
        dropdowns: z.lazy(() => dropdownsSchema),
        openapi: openApiSchema.optional(),
    }),
    baseVersionSchema.extend({
        anchors: z.lazy(() => anchorsSchema),
        openapi: openApiSchema.optional(),
    }),
    baseVersionSchema.extend({
        groups: z.lazy(() => groupsSchema),
        openapi: openApiSchema.optional(),
    }),
    baseVersionSchema.extend({
        pages: z.lazy(() => pagesSchema),
        openapi: openApiSchema.optional(),
    }),
    baseVersionSchema.extend({ openapi: openApiSchema }),
]);
export const decoratedVersionSchema = baseVersionSchema.and(z.union([
    z.object({ href: hrefSchema }),
    z.lazy(() => z.object({ languages: decoratedLanguagesSchema })),
    z.lazy(() => z.object({ tabs: decoratedTabsSchema })),
    z.lazy(() => z.object({ dropdowns: decoratedDropdownsSchema })),
    z.lazy(() => z.object({ anchors: decoratedAnchorsSchema })),
    z.lazy(() => z.object({ groups: decoratedGroupsSchema })),
    z.lazy(() => z.object({ pages: decoratedPagesSchema })),
]));
export const nonRecursiveVersionSchema = baseVersionSchema.extend({ href: hrefSchema });
export const versionsSchema = z
    .array(versionSchema)
    .min(1, 'At least one version must be specified')
    .describe('Organizing by versions');
export const decoratedVersionsSchema = z
    .array(decoratedVersionSchema)
    .min(1, 'At least one version must be specified')
    .describe('Organizing by versions');
